packages/calling-component-bindings/src/hooks/useHandlers.ts

```diff
@@ -1,21 +1,22 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { ReactElement } from 'react';
+import { ReactElement, useContext } from 'react';
 import { StatefulCallClient } from 'calling-stateful-client';
 import { createDefaultCallingHandlersForComponent } from '../handlers/createHandlers';
-import { useCall, useCallAgent, useCallClient, useDeviceManager } from '../providers';
+import { CallClientContext, useCall, useCallAgent, useDeviceManager } from '../providers';
 // @ts-ignore
 import { CommonProperties } from 'acs-ui-common';
 // @ts-ignore
 import { DefaultCallingHandlers } from '../handlers/createHandlers';
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
 export const useHandlers = <PropsT>(component: (props: PropsT) => ReactElement | null) => {
-  const callClient: StatefulCallClient = useCallClient() as any;
+  const callClient: StatefulCallClient = (useContext(CallClientContext) as any)?.callClient;
   const callAgent = useCallAgent();
   const deviceManager = useDeviceManager();
   const call = useCall();
+  if (!callClient) return undefined;
```

Comment:
Should we move this early return up to line 16?

Reply:
Can't. All `use*` calls must be unconditional, so before any early returns.